### ISSUE TITLE
`PLC2801` false positive when `__exit__` is not called

### ISSUE THREAD

**Issue by DetachHead** [astral-sh/ruff#9499]
i have a context manager that wraps another context manager, eg:
```py
from contextlib import AbstractContextManager

class Wrapped:
    def __init__(self, cm: AbstractContextManager[None]):
        self.cm = cm

    def __enter__(self):
        self.cm.__enter__() # PLC2801: Unnecessary dunder call to `__enter__`

    def __exit__(self):
        self.cm.__exit__(None, None, None)
```
[playground](https://play.ruff.rs/21829547-a469-480a-9335-e384a95ed6c2)

in this case, i can't replace the `__enter__` call with a `with` statement because `__exit__` is not called in the same place. ruff should be able to detect cases like this where `__enter__` is called on its own

**Comment by charliermarsh** [comment]
I believe this is fixed by #9496

**Comment by KotlinIsland** [comment]
@charliermarsh this is only resolved within dunder functions, i don't really understand why this is being reported for `__enter__` though:

```py
class A:
    def __enter__(self):
        print("i'm a side effect")

A().__enter__()  # Unnecessary dunder call to `__enter__`
```

i don't really understand this report, and it looks erronious

i have a case where i have a unit test, and i need to `__enter__` an object, but not `__exit__` it, but ruff reports it as an unnecessary dunder call that should be removed

**Comment by DetachHead** [comment]
i think in that case its still correct to report an error. you should never call `__enter__` without also calling `__exit__` in 99% of cases. i think the error message should be reworded though to something more like "call to __enter__ is unsafe. use a `with` statement instead".

the current error message sounds like the call itself does nothing and should be removed, which is not the case.
